src/app/SharpRaven/Data/Context/App.cs

```diff
@@ -0,0 +1,147 @@
+        public DateTimeOffset? StartTime { get; set; }
+        /// <summary>
+        /// Application specific device identifier.
+        /// </summary>
+        [JsonProperty(PropertyName = "device_app_hash", NullValueHandling = NullValueHandling.Ignore)]
+        public string Hash { get; set; }
+        /// <summary>
+        /// String identifying the kind of build, e.g. testflight.
+        /// </summary>
+        [JsonProperty(PropertyName = "build_type", NullValueHandling = NullValueHandling.Ignore)]
+        public string BuildType { get; set; }
+        /// <summary>
+        /// Human readable application name, as it appears on the platform.
+        /// </summary>
+        [JsonProperty(PropertyName = "app_name", NullValueHandling = NullValueHandling.Ignore)]
+        public string Name { get; set; }
+        /// <summary>
+        /// Human readable application version, as it appears on the platform.
+        /// </summary>
+        [JsonProperty(PropertyName = "app_version", NullValueHandling = NullValueHandling.Ignore)]
+        public string Version { get; set; }
+        /// <summary>
+        /// Internal build identifier, as it appears on the platform.
+        /// </summary>
+        [JsonProperty(PropertyName = "app_build", NullValueHandling = NullValueHandling.Ignore)]
+        public string Build { get; set; }
+
+        /// <summary>
+        /// The application AssemblyName used to report Version and Name
+        /// </summary>
+        /// <remarks>
+        /// If no value is set, the entry assemby's name will be used, if possible.
+        /// When not available, a best effort to locate the entrypoint will be made by walking up the stack
+        /// </remarks>
+        [JsonIgnore]
+        public static AssemblyName ApplicationName { get; set; }
+
+        /// <summary>
+        /// Clones this instance
+        /// </summary>
+        /// <returns></returns>
+        internal App Clone()
+        {
+            return new App
+            {
+                Build = this.Build,
+                Version = this.Version,
+                Name = this.Name,
+                BuildType = this.BuildType,
+                Hash = this.Hash,
+                Identifier = this.Identifier,
+                StartTime = this.StartTime
+            };
+        }
+
+        /// <summary>
+        /// Creates an App instance while loading relevant data at runtime
+        /// </summary>
+        /// <returns>An instance if any property was successfuly set or null otherwise.</returns>
+        internal static App Create()
+        {
+            try
+            {
+                var app = new App();
+
+                var asm = ApplicationName ?? EntryAssemblyNameLocator.GetAssemblyName();
+                if (asm != null)
+                {
+                    app.Name = asm.Name;
+                    app.Version = asm.Version?.ToString();
+                }
+
+                using (var proc = Process.GetCurrentProcess())
```

Comment:
Is it wise to `.Dispose()` the current process?

Reply:
Yes. This instance was created by me at this location and should be [disposed when not needed](https://stackoverflow.com/questions/28023490/should-one-call-dispose-for-process-getcurrentprocess).